src/Symfony/Component/Webhook/Client/RequestParser.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Webhook\Client;
+
+use Symfony\Component\HttpFoundation\ChainRequestMatcher;
+use Symfony\Component\HttpFoundation\HeaderBag;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestMatcher\IsJsonRequestMatcher;
+use Symfony\Component\HttpFoundation\RequestMatcher\MethodRequestMatcher;
+use Symfony\Component\HttpFoundation\RequestMatcherInterface;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\RemoteEvent\Event;
+use Symfony\Component\Webhook\Exception\RejectWebhookException;
+
+/**
+ * @author Fabien Potencier <fabien@symfony.com>
+ *
+ * @experimental in 6.3
+ */
+class RequestParser extends AbstractRequestParser
+{
+    public function getSuccessfulResponse(): Response
+    {
+        return new Response('', 202);
+    }
+
+    public function getRejectedResponse(string $reason): Response
+    {
+        return new Response($reason, 406);
+    }
+
+    protected function getRequestMatcher(): RequestMatcherInterface
+    {
+        return new ChainRequestMatcher([
+            new MethodRequestMatcher('POST'),
+            new IsJsonRequestMatcher(),
+        ]);
+    }
+
+    protected function doParse(Request $request, string $secret): Event
+    {
+//$body = $this->bodyParser->parse($request, $secret);
+        $body = $request->toArray();
+
+//$this->headerParser->validate($request, $secret);
+        if (!$request->headers->has('x-id')
+            || !$request->headers->has('x-event')
+            || !$request->headers->has('x-signature')
+        ) {
+            throw new RejectWebhookException(406, 'Missing signature.');
+        }
+
+//$this->signatureParser->validate($request, $secret);
+        $this->validateSignature($request->headers, $request->getContent(), $secret);
+
+        return new Event($request->headers->get('x-event'), $request->headers->get('x-id'), $body);
+    }
+
+    private function validateSignature(HeaderBag $headers, string $body, $secret): void
+    {
+        if (!hash_equals($headers->get('x-signature'), 'sha256='.hash_hmac('sha256', $headers->get('x-event').$headers->get('x-id').$body, $secret))) {
+            throw new RejectWebhookException(406, 'Signature is wrong.');
```

Comment:
Same comment about this error message: would it be useful e.g. to show the value of the expected signature and/or the given one?

Reply:
I don't think we can be more specific here.